File: src/main/java/com/randomperson22/piggymodtest/PiggyModConfigGui.java
package com.randomperson22.piggymodtest;

import java.util.List;

import net.minecraft.client.gui.GuiScreen;
import net.minecraftforge.fml.client.config.GuiConfig;
import net.minecraftforge.fml.client.config.IConfigElement;
import net.minecraftforge.common.config.ConfigElement;
import net.minecraftforge.common.config.Configuration;

public class PiggyModConfigGui extends GuiConfig {
    public PiggyModConfigGui(GuiScreen parent) {
        super(parent, getConfigElements(), "piggymodtest", false, false, "Piggy Mod Test Config");
    }

    private static List<IConfigElement> getConfigElements() {
        return new ConfigElement(PiggyModConfig.config.getCategory(Configuration.CATEGORY_GENERAL)).getChildElements();
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/PiggyModConfigGui.kt
package com.randomperson22.piggymodtest

import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.common.config.ConfigElement
import net.minecraftforge.common.config.Configuration
import net.minecraftforge.fml.client.config.GuiConfig
import net.minecraftforge.fml.client.config.IConfigElement

class PiggyModConfigGui(parent: GuiScreen?) : GuiConfig(
    parent,
    configElements, "piggymodtest", false, false, "Piggy Mod Test Config"
) {
    companion object {
        private val configElements: MutableList<IConfigElement?>?
            get() = ConfigElement(PiggyModConfig.config!!.getCategory(Configuration.CATEGORY_GENERAL))
                .getChildElements()
    }
}
